packages/core/src/module-resolver.ts

```diff
@@ -565,6 +565,7 @@ export class Resolver {
 
   private *componentJSCandidates(inPackageName: string) {
     yield { prefix: '/components/', suffix: '' };
+    yield { prefix: '/components/', suffix: '/index' };
```

Comment:
When I added a similar line to the `componentTemplateCandidates` generator method I ran into an issue with the v2 component. Something about not being allowed to call `setComponentTemplate` twice. My guess is that the v2 component is resolved twice if that scenario, since the extensions logic checks for both .js and .hbs files.

I haven't dug deeper to find the origins of the current setup. Only adding it here made the tests pass and I noticed `yield { prefix: '/components/', suffix: '' };` wasn't added to `componentTemplateCandidates` either, which might be because of similar reasons?